bemani/utils/read.py

```diff
@@ -1620,73 +1648,250 @@ def scrape(self, binfile: str, assets_dir: Optional[str]) -> List[Dict[str, Any]
+                            fp = open(filename, 'rb')
+                            data = fp.read()
+                            fp.close()
+                        else:
+                            fp = open(filename, 'rb')
+                            ifsdata = fp.read()
+                            fp.close()
+                            ifs = IFS(ifsdata)
+                            for fn in ifs.filenames:
+                                _, extension = os.path.splitext(fn)
+                                if extension == '.1':
+                                    data = ifs.read_file(fn)
+
+                        if data is not None:
+                            iidxchart = IIDXChart(data)
+                            bpm_min, bpm_max = iidxchart.bpm
+                            bpm = (bpm_min, bpm_max)
+                            notecounts = iidxchart.notecounts
+                        else:
+                            print(f"Could not find chart information for song {song.id}!")
                     else:
-                        fp = open(filename, 'rb')
-                        ifsdata = fp.read()
-                        fp.close()
-                        ifs = IFS(ifsdata)
-                        for fn in ifs.filenames:
-                            _, extension = os.path.splitext(fn)
-                            if extension == '.1':
-                                data = ifs.read_file(fn)
-
-                    if data is not None:
-                        iidxchart = IIDXChart(data)
-                        bpm_min, bpm_max = iidxchart.bpm
-                        bpm = (bpm_min, bpm_max)
-                        notecounts = iidxchart.notecounts
-                    else:
-                        print(f"Could not find chart information for song {song.id}!")
+                        print(f"No chart information because chart for song {song.id} is missing!")
+                songs.append({
+                    'id': song.id,
+                    'title': song.title,
+                    'artist': song.artist,
+                    'genre': song.genre,
+                    'bpm_min': bpm[0],
+                    'bpm_max': bpm[1],
+                    'difficulty': {
+                        'spn': song.difficulties[0],
+                        'sph': song.difficulties[1],
+                        'spa': song.difficulties[2],
+                        'dpn': song.difficulties[3],
+                        'dph': song.difficulties[4],
+                        'dpa': song.difficulties[5],
+                    },
+                    'notecount': {
+                        'spn': notecounts[0],
+                        'sph': notecounts[1],
+                        'spa': notecounts[2],
+                        'dpn': notecounts[3],
+                        'dph': notecounts[4],
+                        'dpa': notecounts[5],
+                    },
+                })
+
+        qpros: List[Dict[str, Any]] = []
```

Comment:
I should probably have formalized this, including some sort of check, but direct DLL reads are always vulnerable to the wrong datecode DLL being used. I have comments on the Pop'n stuff for exactly what datecode of data should be used in each, maybe we need that here as well?

Reply:
I simply forgot to add those comments so I'll add them shortly